datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -899,6 +914,39 @@ impl PhysicalExpr for InListExpr {
             }
         }
     }
+
+    fn children(&self) -> Vec<Arc<dyn PhysicalExpr>> {
+        let mut children = vec![];
+        children.push(self.expr.clone());
+        children.extend(self.list.clone());
+        children
+    }
+
+    fn with_new_children(
+        self: Arc<Self>,
+        children: Vec<Arc<dyn PhysicalExpr>>,
+    ) -> Result<Arc<dyn PhysicalExpr>> {
+        in_list(
+            children[0].clone(),
+            children[1..].to_vec(),
+            &self.negated,
+            &self.input_schema,
+        )
+    }
+}
+
+impl PartialEq<dyn Any> for InListExpr {
+    fn eq(&self, other: &dyn Any) -> bool {
```

Comment:
This is clever -- it might be worth a comment here explaining that an InListExpr is considered equal if all the elements in the argument list are the same, regardless of order

Reply:
Sure, will add.